### ISSUE TITLE
Make the completion trigger character conditional

### ISSUE THREAD

**Issue by jdneo** [microsoft/vscode-languageserver-node#896]
This is a feature request to make the completion trigger character take effect conditionally.

Considering the following Java code:

```java
String s = new |
```

When user enter the `space`, it would be helpful to show a completion list. This behavior can be observed in other IDEs:

![completion](https://user-images.githubusercontent.com/6193897/154215031-f1266496-790b-4b82-b5da-f60c765a7924.gif)

But simply adding `space` to the trigger characters could be a problem because not every `space` needs to follow a completion action.

Is it possible to make it conditional? For example, only taking effect when there is a `"new"` before `space`.

**Comment by rcjsuen** [comment]
@jdneo I do not believe this is possible. This question/request may be more appropriate for the [microsoft/language-server-protocol repository](https://github.com/microsoft/language-server-protocol/).

**Comment by jdneo** [comment]
@rcjsuen Oh sorry posting to wrong repo.

Please transfer this issue to https://github.com/microsoft/language-server-protocol/
